vsts/windows_csharp.cmd

```diff
@@ -9,20 +9,33 @@ for %%i in ("%build-root%") do set build-root=%%~fi
 
 cd %build-root%
 
-rem Build Debug + run UTs
-call build.cmd -clean -configuration Debug
-if errorlevel 1 goto :eof
+rem Handle Windows Desktop vs Windows IoT Core
+for /f "tokens=3" %%a in ('reg query "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion"  /V EditionID  ^|findstr /ri "REG_SZ"') do set EditionID=%%a
 
-taskkill /F /IM Simulator.exe
-start /D .\vsts\TpmSimulator Simulator.exe
-
-rem Build Release + run E2E Tests
-rem If the build is Delay Signed, this requires strong name validation disabled for our key:
-rem (As Administrator:)  sn -Vr *,31bf3856ad364e35
-rem Re-enable strong name validation for this key using:
-rem (As Administrator:)  sn -Vx
-call build.cmd -clean -configuration Release -e2etests
-if errorlevel 1 goto :eof
+if "%EditionID%"=="IoTUAP" (
+    rem Windows IoT Core: run UTs
+    call build.cmd -clean -configuration Debug -nobuild -nopackage
```

Comment:
In which step do we build the package that is sent to the Win IoT Core image to be tested against?

Reply:
Building is done by the _dotnet test_ command inside the script _build.cmd_ (if no successful build exists, first time building), which is hit regardless of parameter _-nobuild_.
If parameter _-nobuild_ is omitted, the script tries to build for .Net Framework which fails on Windows IoT Core.